tests/includes/server-timing/perflab-server-timing-tests.php

```diff
@@ -53,21 +53,31 @@ public function test_register_metric_stores_metrics_and_runs_measure_callback()
 	}
 
 	public function test_register_metric_runs_measure_callback_based_on_access_cap() {
-		$called = false;
-		$args   = array(
-			'measure_callback' => static function () use ( &$called ) {
-				$called = true;
-			},
-			'access_cap'       => 'manage_options', // Admin capability.
+		$called     = false;
+		$access_cap = 'manage_options'; // Admin capability.
+		$this->server_timing->register_metric(
+			'test-metric',
+			array(
+				'measure_callback' => static function () use ( &$called ) {
+					$called = true;
+				},
+				'access_cap'       => $access_cap,
+			)
 		);
 
-		$this->server_timing->register_metric( 'test-metric', $args );
-
 		$this->assertTrue( $this->server_timing->has_registered_metric( 'test-metric' ), 'Metric without cap should still be registered' );
 		$this->assertFalse( $called, 'Measure callback without cap must not be run' );
 
 		wp_set_current_user( self::$admin_id );
-		$this->server_timing->register_metric( 'test-metric-2', $args );
```

Comment:
Without passing the callback here, PHPStan isn't aware that the `$called` variable could actually have changed so it complains:

> Call to method PHPUnit\Framework\Assert::assertTrue() with false and 'Measure callback…' will always evaluate to false.